learning_resources_search/serializers.py

```diff
@@ -10,10 +16,217 @@
     gen_course_id,
     gen_program_id,
 )
+from learning_resources_search.constants import AGGREGATIONS
 
 log = logging.getLogger()
 
 
+def extract_values(obj, key):
+    """
+    Pull all values of specified key from nested JSON.
+
+    Args:
+        obj(dict): The JSON object
+        key(str): The JSON key to search for and extract
+
+    Returns:
+        list of matching key values
+
+    """
+    array = []
+
+    def extract(obj, array, key):
+        """Recursively search for values of key in JSON tree."""
+        if isinstance(obj, dict):
+            for k, v in obj.items():
+                if k == key:
+                    array.append(v)
+                if isinstance(v, dict | list):
+                    extract(v, array, key)
+        elif isinstance(obj, list):
+            for item in obj:
+                extract(item, array, key)
+        return array
+
+    return extract(obj, array, key)
+
+
+class StringArrayField(ListField):
+    """
+    Character separated ListField.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def to_internal_value(self, data):
+        return ",".join(data)
+
+    def to_representation(self, data):
+        return data.split(",")
```

Comment:
I think we should try to leverage more of Django's serializers for two reasons:
1. Let django's own serializers handle more validation
2. better openapi generation

For example, with
```
resource_type = ListSerializer(
    required=False,
    child=ChoiceSerializer(
        choices = [e.value.lower() for e in LearningResourceType]
    )
)
```
drf-spectacular will produce a list of choices for the value automatically using OpenAPI's `enum` field, rather than relying on a `help_text`.

**Note:** Using `ListSerializer` directly (rather than `StringArrayField`) means that query parameter arrays must passed with `&`: `?foo=a,b,c` becomes `?foo=a&foo=b&foo=c`. (In OpenAPI lingo, `?foo=a&foo=b&foo=c` is `explode: true`.). If we want to keep the comma-separated version, I think we could do
```
    def to_internal_value(self, data):
        return super().to_representation(data[0].split(","))
```